src/messaging.rs

```diff
@@ -25,6 +26,32 @@ pub enum Target<N> {
     Node(N),
 }
 
+impl<N: Ord> Ord for Target<N> {
+    fn cmp(&self, other: &Target<N>) -> Ordering {
+        match (self, other) {
+            (Target::Node(id1), Target::Node(id2)) => id1.cmp(id2),
+            (Target::All, Target::All) => Ordering::Equal,
+            (Target::All, Target::Node(_)) => Ordering::Greater,
+            (Target::Node(_), Target::All) => Ordering::Less,
+        }
```

Comment:
Can't this be derived?

Reply:
In fact it's not used. So we can simply remove it.